modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java

```diff
@@ -361,7 +362,14 @@ public String extractIndexableAttributes(
 
 		Fields fields = _toFields(ddmStructure, ddmFormValues);
 
-		for (Field field : fields) {
+		Map<String, List<DDMFormFieldValue>> ddmFormFieldValuesMap =
+			ddmFormValues.getDDMFormFieldValuesMap(true);
+
+		for (Map.Entry<String, List<DDMFormFieldValue>> entry :
+				ddmFormFieldValuesMap.entrySet()) {
```

Comment:
@marcelabc can we use `keySet()` instead of `entrySet()` here? i.e. `for (String key : ddmFormFieldValuesMap.keySet())`.